cli/cli_app.go

```diff
@@ -451,15 +452,23 @@ func runTerragrunt(terragruntOptions *options.TerragruntOptions) (finalStatus er
 		}
 	}
 
-	if err := conf.ImportFiles.Run(err, foldersWithTerraformFiles...); stopOnError(err) {
-		return
+	if folders := foldersWithTerraformFiles[1:]; len(folders) > 0 {
+		if err := conf.ImportFiles.Run(err, folders...); stopOnError(err) {
+			return
+		}
 	}
 
 	// Run Gotemplate
 	if err == nil && useTempFolder && terragruntOptions.ApplyTemplate {
 		template.TemplateLog.SetDefaultConsoleHookLevel(terragruntOptions.Logger.GetLevel())
+		template.InternalLog.SetDefaultConsoleHookLevel(terragruntOptions.Logger.GetLevel() - 1)
 		var t *template.Template
-		if t, err = template.NewTemplate(terragruntOptions.WorkingDir, terragruntOptions.GetContext(), "", nil); stopOnError(err) {
+		subst := []string{
+			`/(?m)^(?P<attr>\s*provider\s*=\s*)"(?P<name>.*)"\s*$/$attr$name`,
```

Comment:
What about this. What does it do? It needs a comment too

Reply:
Regarding the substitutions. Sorry, I splitted my (as usual) too big PR into smaller PRs and I forgot to isolate that part. The goal is to remove terraform warnings when we generate code that is not 100% terraform 0.12 compliant such as:
`provider = "aws.@default_region"
that will become:
`provider = "aws.us-east-1"`
and will be transformed by this substitution to:
`provider = aws.us-east-1`

Reply:
It's fine. It just needs a comment and a test